src/main/java/hudson/plugins/mercurial/MercurialSCM.java

```diff
@@ -352,7 +353,8 @@ public SCMRevisionState calcRevisionsFromBuild(Run<?, ?> build, FilePath workspa
         try {
         String tip = hg.tip(workspace2Repo(workspace, env), null);
         String rev = hg.tipNumber(workspace2Repo(workspace, env), null);
-        return tip != null && rev != null ? new MercurialTagAction(tip, rev, getSubdir(env)) : null;
+        String branch = revisionType == RevisionType.TAG ? hg.branch(workspace2Repo(workspace, env), null) : null;
```

Comment:
For future expansion of `RevisionType` this should really be `revisionType != RevisionType.BRANCH`, but for now it does not matter.